text/0000-shared-layers.md

```diff
@@ -0,0 +1,117 @@
+
+One example can be a CCache buildpack that sets up CCACHE_DIR environment variable (pointing to a directory in a shared layer), which a CMake buildpack may use for its build cache.
+
+Another example can be a pip cache buildpack that sets up a PIP_CACHE_DIR environment variable (pointing to a directory in a shared layer), which could be used by a [pip-tools](https://github.com/jazzband/pip-tools) buildpack, or a [pipenv](https://pipenv.pypa.io/en/latest/) buildpack to resolve the python dependency tree and generate a requirements.txt and subsequently the pip buildpack could install these requirement using the wheels cached during the dependency resolution.
+
+- What is the expected outcome?
+
+The spec and lifecycle is modified to support the above use cases.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This RFC introduces a new top level folder structure called `<shared-layers>` which shall live on the same level as the `<layers>` directory.
+
+The structure would look like - 
+
+```
+<shared-layers> (configurable with `CNB_SHARED_LAYERS_DIR`)
+└── <escaped-buildpack-id> `CNB_BP_SHARED_LAYERS_DIR`
+    ├── <shared-layer>.toml
+    └── <shared-layer>
+        ├── <usable-directory>
+        └── env
+```
+
+Each buildpack will be passed an environment variable `CNB_BP_SHARED_LAYERS_DIR` with the location to the shared layers directory for a buildpack.
+
+The `<shared-layer>.toml` would look like - 
+
+```toml
+[types]
+  cache = false
+
+[metadata]
+# buildpack-specific data
+```
+
+The `env` directory would be used to set build time only variables for subsequent buildpacks. It will follow an API similar to [Buildpack ENV API](https://github.com/buildpacks/spec/blob/main/buildpack.md#provided-by-the-buildpacks)
+
+
+# How it Works
+[how-it-works]: #how-it-works
+
+All layers created in the `shared-layers` directory are build layers by default. The `cache` layer flag decides if the layer will be cached for the next run or not. This flag will follow the same semantics as [specified here.](https://github.com/buildpacks/spec/blob/main/buildpack.md#cached-layers). The `<shared-layer>.toml` for each layer will be restored with the same semantics as [specified here.](https://github.com/buildpacks/spec/blob/main/buildpack.md#cached-layers) with the `types` table being omitted during subsequent restores to avoid stale layers.
+
+A `shared-layer` is associated with a specific buildpack that created it and it is expected to be wiped if the associated buildpack is not involved during the build process. The `env` directory is co-located with the `shared-layer` so that the associated environment variables that may point to directories in the `shared-layer` follow the same lifecycle as the `shared-layer` itself.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+- Added maintanence complexity.
+- Additional complexity for buildpack authors to account for.
+
+# Alternatives
```

Comment:
Should we list an alternative that is a simple build cache dumping ground (aka not owned by a specific buildpack)? Whatever is there is always restored, no automatic cleanup, etc.